cromwell_tools/_cromwell_auth.py

```diff
@@ -0,0 +1,91 @@
+import json
+import requests
+import requests.auth
+from oauth2client.service_account import ServiceAccountCredentials
+from ._cromwell_api import CromwellAPI
+
+
+class AuthenticationError(Exception):
+    pass
+
+
+class CromwellAuth:
+
+    def __init__(self, url, header, auth):
+        """Authentication Helper for a Cromwell Server
+
+        :param dict header: authorization header
+        """
+        if not header and not auth:
+            raise ValueError("either header or auth must be passed")
+
+        if header is not None:
+            if not isinstance(header, dict):
+                raise TypeError('if passed, header must be a dict')
+            if "Authorization" not in header.keys():
+                raise TypeError('the header must have an "Authorization" key')
+        self.header = header
+
+        # what is the type and what checks can be run?
+        self.auth = auth
+
+        if isinstance(url, str) and 'http' in url:
```

Comment:
`url.startswith(`http`)` is a slightly more reliable than checking if the url contains `http`. I think there are url validator libraries out there as well that would be even better.